ci/check-protobuf-stability.sh

```diff
@@ -89,26 +89,33 @@ USAGE
   exit
   ;;
 --stable)
-  if [[ $TARGET == "main" ]]; then
-    # Check against the most recent stable tag.
-    # This check runs only on main or PRs targeting main.
-    #
-    # This check does not need to run on release branch commits because
-    # they are built sequentially, so no conflicts are possible and the per-PR
-    # check is enough.
-    readonly LATEST_STABLE_TAG="$(git tag | grep -v "snapshot" | sort -V | tail -1)"
-    # The v1.17 stable release includes the buf config file with the default name `buf.yml`.
-    # Starting with v1.18 we have multiple buf config files.
-    if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
-      BUF_CONFIG_UPDATED=false
-    else
-      BUF_CONFIG_UPDATED=true
-    fi
-    BUF_GIT_TARGET_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
-    check_protos
+  # Check against the highest stable tag.
+  #
+  # This check does not need to run on release branch commits because
+  # they are built sequentially, so no conflicts are possible and the per-PR
+  # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
+  LATEST_STABLE_TAG=""
+  if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
+    readonly VERSION="${TARGET#release/}"
+    readonly VERSION_PREFIX="${VERSION%.x}"
+    readonly STABLE_TAGS=($(git tag | grep "v.*" | grep -v "snapshot" | sort -V))
+    LATEST_STABLE_TAG="$(for TAG in "${STABLE_TAGS[@]}"; do if [[ $(semver compare "${TAG#v}" "${VERSION_PREFIX}.999") == "-1" ]]; then echo "$TAG"; fi; done | tail -1)"
```

Comment:
For better readability: I'd split this into multiple lines and I'd bind `${TAG#v}`. Also, I don't love the `.999` hack; how about  using the next minor for the comparison instead, since equality is excluded anyway?

Reply:
FYI Bash has no issue with multiple lines within a subshell (`$()`).

Reply:
@fabiotudone-da, sorry, next minor to what? What we get here is `1.17.x` for example.

Reply:
@fabiotudone-da, WDYT about extracting it as a `MAX_MINOR_VERSION` variable?

Reply:
Or, you'd like to get the next version to `1.17` (with `semver bump`) which is `1.18` and compare to it. Do I get it correctly?

Reply:
> What we get here is 1.17.x for example.

afaiu, an example would be, if we have a PR targeting `release/1.6.x`, `semver compare 1.16.1 1.16.999`. Why not using `semver compare 1.16.1 1.17`?

Reply:
> WDYT about extracting it as a MAX_MINOR_VERSION variable?

For `${TAG#v}`? I was more thinking of `CURRENT_STABLE_VERSION` or something like that.

Reply:
Ok, sorry for the spam, I think we're now on the same page. I made relevant changes.